chef/cookbooks/neutron/recipes/l3.rb

```diff
@@ -156,6 +156,16 @@
   )
 end
 
+template "/etc/neutron/dnsmasq-neutron.conf" do
+  source "dnsmasq-neutron.conf.erb"
+  owner "root"
+  group "root"
+  mode "0644"
+  variables(
+    :mtu => node[:neutron][:ml2_type_drivers].include?('gre') ? 1400 : 1500
```

Comment:
You're still missing the "vxlan" case here. We need the lower MTU in that case as well.